forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/api/accesstoken/AccessTokenApiController.java

```diff
@@ -23,82 +23,83 @@
 import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.security.core.Authentication;
-import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Controller;
 import org.springframework.util.MultiValueMap;
 
 import javax.servlet.http.HttpServletRequest;
-import java.nio.charset.Charset;
 import java.security.Principal;
-import java.text.ParseException;
-import java.util.Base64;
 import java.util.UUID;
 
 @Controller
 @Slf4j
 public class AccessTokenApiController implements AccessTokenApi {
 
-    private AMASPSPGateway amGateway;
-    private HeadLessAccessTokenService headLessAccessTokenService;
-    private TppStoreService tppStoreService;
-    private AMGatewayService amGatewayService;
-    private JwtOverridingService jwtOverridingService;
-    private TokenUsageService tokenUsageService;
+    private final AMASPSPGateway amGateway;
+    private final HeadLessAccessTokenService headLessAccessTokenService;
+    private final AMGatewayService amGatewayService;
+    private final JwtOverridingService jwtOverridingService;
+    private final MatlsRequestVerificationService matlsRequestVerificationService;
 
-    private static final String CLIENT_ASSERTION = "client_assertion";
     private static final String GRANT_TYPE = "grant_type";
-    private static final String CLIENT_ID = "client_id";
-    private static final String BASIC = "Basic";
+
 
     @Autowired
     public AccessTokenApiController(AMASPSPGateway amGateway, HeadLessAccessTokenService headLessAccessTokenService,
-                                    TppStoreService tppStoreService, AMGatewayService amGatewayService,
-                                    JwtOverridingService jwtOverridingService, TokenUsageService tokenUsageService) {
+                                    AMGatewayService amGatewayService, JwtOverridingService jwtOverridingService,
+                                    MatlsRequestVerificationService matlsRequestVerificationService) {
         this.amGateway = amGateway;
         this.headLessAccessTokenService = headLessAccessTokenService;
-        this.tppStoreService = tppStoreService;
         this.amGatewayService = amGatewayService;
         this.jwtOverridingService = jwtOverridingService;
-        this.tokenUsageService = tokenUsageService;
+        this.matlsRequestVerificationService = matlsRequestVerificationService;
     }
 
     @Override
     public ResponseEntity getAccessToken(MultiValueMap paramMap, String authorization, Principal principal,
                                          HttpServletRequest request) throws OBErrorResponseException, OBErrorException {
-
-        PairClientIDAuthMethod clientIDAuthMethod = verifyMATLSMatchesRequest(paramMap, authorization, principal);
+        log.debug("getAccessToken(), paramMap {}", paramMap);
```

Comment:
Would MultiValueMap benefit from being Parameterized?

Reply:
It would be nice if it was.